app.go

```diff
@@ -81,9 +83,21 @@ func (f optionFunc) apply(app *App) { f(app) }
 //
 // See the documentation for go.uber.org/dig for further details.
 func Provide(constructors ...interface{}) Option {
-	return optionFunc(func(app *App) {
-		app.provides = append(app.provides, constructors...)
-	})
+	return provideOption(constructors)
+}
+
+type provideOption []interface{}
+
+func (po provideOption) apply(app *App) {
+	app.provides = append(app.provides, po...)
```

Comment:
Can we change the Option interface to return errors on apply?
So in that case we'll return an error if `po` is not a collection of functions.
As I've mentioned in https://github.com/uber-go/fx/pull/562,
we should try to detect errors earlier, during construction time.